ceremonies/src/tests.rs

```diff
@@ -1804,3 +1807,134 @@ fn set_location_tolerance_works() {
+		let endorsee = account_id(&AccountKeyring::Ferdie.pair());
+		let unregistered_user = account_id(&AccountKeyring::Charlie.pair());
+
+		assert!(EncointerBalances::issue(cid, &reputable, NominalIncome::from_num(1)).is_ok());
+		cindex += 1;
+
+		run_to_next_phase();
+		run_to_next_phase();
+		run_to_next_phase();
+
+		assert_ok!(EncointerCeremonies::endorse_newcomer(
+			Origin::signed(bootstrapper.clone()),
+			cid,
+			endorsee.clone()
+		));
+
+		assert_ok!(EncointerCeremonies::register(cid, cindex, &bootstrapper, false));
+		assert_ok!(EncointerCeremonies::register(cid, cindex, &reputable, true));
+		assert_ok!(EncointerCeremonies::register(cid, cindex, &newbie, false));
+		assert_ok!(EncointerCeremonies::register(cid, cindex, &endorsee, false));
+
+		assert_eq!(
+			EncointerCeremonies::get_participant_type((cid, cindex), &bootstrapper),
+			Some(ParticipantType::Bootstrapper)
+		);
+
+		assert_eq!(
+			EncointerCeremonies::get_participant_type((cid, cindex), &newbie),
+			Some(ParticipantType::Newbie)
+		);
+
+		assert_eq!(
+			EncointerCeremonies::get_participant_type((cid, cindex), &endorsee),
+			Some(ParticipantType::Endorsee)
+		);
+
+		assert_eq!(
+			EncointerCeremonies::get_participant_type((cid, cindex), &reputable),
+			Some(ParticipantType::Reputable)
+		);
+
+		assert_eq!(
+			EncointerCeremonies::get_participant_type((cid, cindex), &unregistered_user),
+			None
+		);
+	});
+}
+
+#[test]
+fn get_aggregated_account_data_works() {
+	new_test_ext().execute_with(|| {
+		let cid = register_test_community::<TestRuntime>(None, 1.0, 1.0);
+		let cindex = EncointerScheduler::current_ceremony_index();
+
+		let bootstrapper = account_id(&AccountKeyring::Alice.pair());
+		let bootstrapper2 = account_id(&AccountKeyring::Bob.pair());
+		EncointerCommunities::insert_bootstrappers(
+			cid,
+			vec![bootstrapper.clone(), bootstrapper2.clone()],
+		);
+
+		let reputable = account_id(&AccountKeyring::Ferdie.pair());
+		let reputable2 = account_id(&AccountKeyring::Charlie.pair());
+
+		assert!(EncointerBalances::issue(cid, &reputable, NominalIncome::from_num(1)).is_ok());
+
+		assert_ok!(EncointerCeremonies::register(cid, cindex, &bootstrapper, false));
+		assert_ok!(EncointerCeremonies::register(cid, cindex, &bootstrapper2, false));
+
+		assert_ok!(EncointerCeremonies::register(cid, cindex, &reputable, true));
+
+		let mut aggregated_account_data =
+			EncointerCeremonies::get_aggregated_account_data(cid, &bootstrapper);
+
+		assert_eq!(aggregated_account_data.global.ceremony_phase, CeremonyPhaseType::REGISTERING);
+		assert_eq!(aggregated_account_data.global.ceremony_index, 1);
+		let mut personal = aggregated_account_data.personal.unwrap();
+		assert_eq!(personal.participant_type, ParticipantType::Bootstrapper);
+		assert_eq!(personal.meetup_index, None);
+		assert_eq!(personal.meetup_location_index, None);
+		assert_eq!(personal.meetup_time, None);
+		assert_eq!(personal.meetup_registry, None);
```

Comment:
I don't really understand this case, how can this happen the way I read the code this should fail. Can you help me here? :)

Reply:
it is the case when the participant is registered, but the assignment has not yet been made

Reply:
Ah, ok that makes sense!